src/util/to-rmg.ts

```diff
@@ -0,0 +1,575 @@
+    one_line: true,
+    int_padding: 355,
+};
+
+const useStn: any = {};
+
+const newParamTemple = {
+    svgWidth: {
+        destination: 1500,
+        runin: 1500,
+        railmap: 1500,
+        indoor: 1500,
+    },
+    svg_height: 400,
+    style: 'shmetro',
+    y_pc: 50,
+    padding: 10,
+    branchSpacingPct: 33,
+    direction: 'r',
+    platform_num: '1',
+    theme: ['hongkong', 'twl', '#E2231A', '#fff'],
+    line_name: ['地鐵線', 'Metro Line'],
+    current_stn_idx: 'jlaMj2',
+    stn_list: useStn,
+    namePosMTR: {
+        isStagger: true,
+        isFlip: true,
+    },
+    customiseMTRDest: {
+        isLegacy: false,
+        terminal: false,
+    },
+    line_num: '1',
+    psd_num: '1',
+    info_panel_type: 'gz1',
+    notesGZMTR: [],
+    direction_gz_x: 40,
+    direction_gz_y: 70,
+    coline: {},
+    loop: false,
+    loop_info: {
+        bank: true,
+        left_and_right_factor: 0,
+        bottom_factor: 1,
+    },
+};
+
+const newRMGStn: RMGStn = {
+    name: [],
+    secondaryName: false,
+    num: '',
+    services: ['local'],
+    parents: [],
+    children: [],
+    branch: {
+        left: [],
+        right: [],
+    },
+    transfer: {
+        groups: [
+            {
+                lines: [],
+            },
+        ],
+        tick_direc: 'r',
+        paid_area: true,
+    },
+    facility: '',
+    loop_pivot: false,
+    one_line: true,
+    int_padding: 355,
+};
+
+// convert color['shanghai', 'sh1', ...] to a string (for compare)
+const colorToString = (color: Array<any>) => {
+    return String(color[0] + '/' + color[1] + '=' + color[2] + color[3]);
+};
+
+// reverse an array
+const reverse = (a: Array<any>) => {
+    const p = [];
+    for (let i = a.length - 1; i >= 0; i--) {
+        p.push(a[i]);
+    }
+    return p;
+};
+
+// verify the line whether is needed to add
+const isColorLine = (type: any) => {
+    if (type == LineStyleType['SingleColor'] || type == LineStyleType['MTRRaceDays']) {
+        return true;
+    } else return false;
+};
+
+// get line color array
+const getColor = (attr: EdgeAttributes) => {
+    let nowColor = new Array<string>();
+    if (attr.style == LineStyleType['SingleColor']) {
+        const newAttr = attr[LineStyleType['SingleColor']] as SingleColorAttributes;
+        nowColor = newAttr.color;
+    } else if (attr.style == LineStyleType['MTRRaceDays']) {
+        const newAttr = attr[LineStyleType['MTRRaceDays']] as SingleColorAttributes;
+        nowColor = newAttr.color;
+    }
+    return structuredClone(nowColor);
+};
+
+// add edge
+const addEdge = (graph: MultiDirectedGraph<NodeAttributes, EdgeAttributes, GraphAttributes>, lineId: string) => {
+    const u = graph.extremities(lineId)[0];
+    const v = graph.extremities(lineId)[1];
+    const now = graph.getEdgeAttributes(lineId);
+    const nowStyle: string = now.style;
+    if (u == v) return; // skip u-u
+    if (isColorLine(nowStyle)) {
+        const nowColor = getColor(now);
```

Comment:
What will happen if `nowColor` is `[]` after `getColor(now)`?

Reply:
It won't get a `[]` from `getColor()` if we give a legal line attributes `now`.

I think you may need to  add the following to make sure it is not `[]`

```typescript
if (nowColor == []) return;
```

Reply:
So as long as the graph is valid, it will never be [] right?

Reply:
Yep.